exercises-python/rachel_baust/ex01_basics/main.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+from re import split
+from os.path import exists
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Count words from a file')
+    parser.add_argument('filename', help='The path to the file')
+    args = parser.parse_args()
+
+    file_content = read_content(args.filename)
+    if file_content is not None:
+        word_count = create_word_dictionary(file_content)
+        print_list('Word   Usage', sorted(word_count.items(), key=lambda x: x[1], reverse=True))
+
+
+def read_content(filename):
+    """
+    Reads in all the lines from the file at the given path
+    :param filename: the path to the file to be read
+    :return: list[str] or None if the file couldn't be found
+    """
+    if exists(filename):
```

Comment:
Ah, thanks - I'd started off playing around with exceptions but got into a bit of tangle because I was using a more long-winded `file.open()`, `file.close()` approach to begin with. The exceptions seem to be working nicely with the way the code is now, though.